httpcore5/src/main/java/org/apache/hc/core5/net/URIAuthority.java

```diff
@@ -87,23 +64,8 @@ static URIAuthority parse(final CharSequence s, final Tokenizer.Cursor cursor) t
             //Rewind
             cursor.updatePos(initPos);
         }
-        final String hostName = tokenizer.parseContent(s, cursor, PORT_SEPARATORS);
-        String portText = null;
-        if (!cursor.atEnd() && s.charAt(cursor.getPos()) == ':') {
-            cursor.updatePos(cursor.getPos() + 1);
-            portText = tokenizer.parseContent(s, cursor, TERMINATORS);
-        }
-        final int port;
-        if (!TextUtils.isBlank(portText)) {
-            try {
-                port = Integer.parseInt(portText);
-            } catch (final NumberFormatException ex) {
-                throw createException(s, cursor, "Authority port is invalid");
-            }
-        } else {
-            port = -1;
-        }
-        return new URIAuthority(userInfo, hostName.toLowerCase(Locale.ROOT), port, true);
+        final Host host = Host.parse(s, cursor);
+        return new URIAuthority(userInfo, host.getHostName(), host.getPort());
```

Comment:
Stupid question, why can't a `URIAuthority` be backed by a  `Host` just like `HttpHost` does?

Reply:
@michael-o I am not sure I understand what you mean. Why does not `URIAuthority` extend `Host`?

Reply:
Not extend, but `URIAuthority(UserInfo, Host)` instead of `URIAuthority(UserInfo, String, int)`. Thus, not maintain the raw fields internally, but use `Host`. Delegation, not extension.

Reply:
@michael-o No reason, actually. This can be easily done.

Reply:
@michael-o Please take a look.

Reply:
I'd rather prefer to pass the `Host` instance directly and get rid of the actual private fields.

Reply:
@michael-o Please take another look.

Reply:
Almost perfect, just pass the `Host` instance directly instead of having another instance created by the constructor.

Reply:
> Almost perfect, just pass the `Host` instance directly instead of having another instance created by the constructor.

@michael-o I already do, don't I? See line 107

Reply:
Yes, you do, but here
https://github.com/apache/httpcomponents-core/blob/d8296ab690f7d45b6d659878e86a5c870f2d00db/httpcore5/src/main/java/org/apache/hc/core5/net/URIAuthority.java#L66-L67
you use the getters and create new object here again
https://github.com/apache/httpcomponents-core/blob/d8296ab690f7d45b6d659878e86a5c870f2d00db/httpcore5/src/main/java/org/apache/hc/core5/net/URIAuthority.java#L94-L98

Reply:
@michael-o Corrected.

Reply:
Very nice now!